app/presenters/tree_builder_menu_roles.rb

```diff
@@ -0,0 +1,155 @@
+class TreeBuilderMenuRoles < TreeBuilder
+  has_kids_for Hash, [:x_build_kids]
+
+  attr_reader :rpt_menu, :role_choice
+
+  def initialize(name, type, sandbox, role_choice, rpt_menu = nil)
+    @rpt_menu    = rpt_menu || sandbox[:rpt_menu]
+    @role_choice = role_choice
+
+    super(name, type, sandbox, true)
+  end
+
+  # Used for testing convenience and to satisfy need for
+  # tree data in controller
+
+  def hash_tree
+    @hash_tree ||= x_build_tree(@tree_state.x_tree(@name))
+  end
+
+  private
+
+  def set_locals_for_render
+    {
+      :tree_id       => "#{@name}box",
+      :tree_name     => @name.to_s,
+      :bs_tree       => @bs_tree,
+      :click_url     => "/report/menu_editor/",
+      :onclick       => "miqMenuEditor",
+      :cookie_prefix => "edit_",
+      :tree_state    => true,
+      :checkboxes    => false
+    }
+  end
+
+  # Set add_root to false, otherwise we need a root_options method
+  #
+  def tree_init_options(_tree_name)
+    { :lazy => false, :add_root => false }
+  end
+
+  # TreeBuilder Overrides
+
+  # Override this method to add our custom root.
+  # Our root (Top Level) needs a specific id to make the form work.
+  #
+  def x_build_tree(options)
+    # Tell super method to always give us child_nodes only.
+    options[:add_root] = false
+    nodes = super(options)
+
+    [x_build_actual_root_node(nodes)]
+  end
```

Comment:
There doesn't seem to be another way to add a root node with specific attributes. The "Top Level" folder needs a specific id to work, so `root_options` wasn't cutting it.